pkg/snapshotmgr/snapshot_manager.go

```diff
@@ -183,30 +184,83 @@ func (this *SnapshotManager) CreateSnapshot(peID astrolabe.ProtectedEntityID, ta
 
 func (this *SnapshotManager) DeleteSnapshot(peID astrolabe.ProtectedEntityID) error {
 	log := this.WithField("peID", peID.String())
-	log.Infof("Step 1: Deleting the local snapshot")
-	err := this.DeleteLocalSnapshot(peID)
+	log.Infof("Step 0: Cancel on-going upload.")
+	config, err := rest.InClusterConfig()
 	if err != nil {
-		log.WithError(err).Errorf("Failed to delete the local snapshot for peID")
-	} else {
-		log.Infof("Deleted the local snapshot")
+		this.WithError(err).Errorf("Failed to get k8s inClusterConfig")
+		return err
 	}
-
-	isLocalMode := utils.GetBool(this.config[utils.VolumeSnapshotterLocalMode], false)
-
-	if isLocalMode {
-		return nil
+	pluginClient, err := plugin_clientset.NewForConfig(config)
+	if err != nil {
+		this.WithError(err).Errorf("Failed to get k8s clientset from the given config: %v ", config)
+		return err
 	}
 
-	log.Infof("Step 2: Deleting the durable snapshot from s3")
-	err = this.DeleteRemoteSnapshot(peID)
+	// look up velero namespace from the env variable in container
+	veleroNs, exist := os.LookupEnv("VELERO_NAMESPACE")
+	if !exist {
+		this.WithError(err).Errorf("DeleteSnapshot: Failed to lookup the env variable for velero namespace")
+		return err
+	}
+	uploadName := "upload-" + peID.GetSnapshotID().GetID()
+	log.Infof("Searching for Upload CR: %v", uploadName)
+	uploadCR, err := pluginClient.VeleropluginV1().Uploads(veleroNs).Get(uploadName, metav1.GetOptions{})
 	if err != nil {
-		if !strings.Contains(err.Error(), "The specified key does not exist") {
-			log.WithError(err).Errorf("Failed to delete the durable snapshot for PEID")
+		log.WithError(err).Errorf(" Error while retrieving the upload CR %v", uploadName)
+	}
+	// An upload is considered done when it's in either of the terminal stages- Completed, CleanupFailed, Canceled
+	uploadCompleted := this.isTerminalState(uploadCR)
+	if uploadCR != nil && !uploadCompleted {
+		log.Infof("Found the Upload CR: %v, updating spec to indicate cancel upload.", uploadName)
+		timeNow := clock.RealClock{}
+		mutate := func(r *v1api.Upload) {
+			uploadCR.Spec.UploadCancel = true
+			uploadCR.Status.Phase = v1api.UploadPhaseCanceling
```

Comment:
Let's make the canceling phase be what the data manager sets when it starts processing the cancel.  I want to be able to see if the data manager is paying attention to the cancel request.  If it never moves to UploadPhaseCanceling then we know that the Data Manager is having a problem seeing the CR.  If it gets stuck in UploadPhaseCanceling then we know it has a problem stopping the upload.